### ISSUE TITLE
Question about layer ordering in the UI

### ISSUE THREAD

**Issue by colinbeier** [caryfitzhugh/NYCCSC-map-viewer#108]
NYSERDA wanted to know if there were other options for reorganizing the active layers in the UI.  They recognize the up arrow for pushing layers to the top, and I reminded them that turning a layer off, then back on, will put that layer on the top as well.  

NYSERDA was wondering - not requesting - if it were possible to have a 'drag and drop' type of UI where a layer name in the legend tab could be grabbed and pulled into a particular place in order.  Amanda Stevens (our PM at NYSERDA) would like it to "work like her Netflix queue" if possible.  

I explained to them that while such a UI feature would be snazzy, it really wouldn't add much functionality, because one can only stack so many layers in a map before it becomes too busy to work with.  The layer cloning, opacity and hide/view functions do more of the work in this regard than just the ordering of the layers.  

Basically, unless this is very easy to do, we need to tell them that it will take too much time and money for a relatively small return in terms of functionality.  Just wanted to get your input to be sure.

**Comment by caryfitzhugh** [comment]
hey colin.  
I could spend 1/2 an hour and investigate.  I don't think it would take more than a few hours at most.  Probably more like 1 hour.

**Comment by colinbeier** [comment]
Not sure if even that amount of time is worth the effort... maybe look into it (briefly) but let's hold off on anything further until I can consult with Lisa.

Colin Beier
SUNY ESF

On Jan 11, 2016 12:56 PM, Cary FitzHugh <notifications@github.com> wrote:

hey colin.

I could spend 1/2 an hour and investigate. I don't think it would take more than a few hours at most. Probably more like 1 hour.

**Comment by caryfitzhugh** [comment]
well - I investigated, and it was really easy! . done.

**Comment by caryfitzhugh** [comment]
doh. there is one trouble - the drag event clobbers when you drag an input slider.

**Comment by sasignell** [comment]
this probably isn't worth the trouble then

On Mon, Jan 11, 2016 at 2:17 PM, Cary FitzHugh <notifications@github.com>
wrote:

> doh. there is one trouble - the drag event clobbers when you drag an input
> slider.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/108#issuecomment-170659541>
> .
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by colinbeier** [comment]
yea, i agree, don't go down a rabbit hole for this... not even a mole hole.  :)

**Comment by caryfitzhugh** [comment]
that wasn't too bad. should be done now.  for about the 1/2 hour I was gonig to take investigating.

**Comment by colinbeier** [comment]
is it pushed to the gh pages so i can take a look-see?

**Comment by caryfitzhugh** [comment]
not yet. once sec.

**Comment by caryfitzhugh** [comment]
http://caryfitzhugh.github.io/NYCCSC-map-viewer/dist/map_viewer.html

**Comment by caryfitzhugh** [comment]
grab the hamburgers.
it's a little wonky if you have some opena dn some closed layers...  but that's a rabbit hole / mole hole.
the only real solutions are to, when you start dragging, collapse or expand everything...

I guess you could then (once done), re-expand things which were expanded before, but then you need to re-scroll to the new... bleck.

**Comment by colinbeier** [comment]
nice!

**Comment by colinbeier** [comment]
well it does seem to bump the point layers to the top, at least in the tab, every time you expand or collapse a layer.  but they are always on top, right?  doing this with the air monitoring stations.  the rest of the layers that are not point layers stay in the order arranged.  interesting.  

we can leave it as-is and see if they want to pursue any further tweaks.  not bad for a half-hour -- it will make the boss happy!

**Comment by caryfitzhugh** [comment]
oh oh... that's interesting. I should change that. I forgot about those layers! Jus tnee to turn off the sorting there.  Done. Deploying in a moment to gh-pages with the narccap precip layer and 10-yr bin changes, etc.
